comms/core/src/connection_manager/peer_connection.rs

```diff
@@ -315,30 +315,36 @@ impl PeerConnection {
         self.force_disconnect_rpc_clients_when_clone_drops
             .store(true, Ordering::Relaxed);
     }
+
+    /// Explicitly cleanup RPC clients to avoid circular reference issues in Drop
```

Comment:
this comment does not make sense, its just the same code as before, now called in a separate fn?